save-backend/src/main/kotlin/com/saveourtool/save/backend/event/VulnerabilityMetadataListener.kt

```diff
@@ -0,0 +1,46 @@
+package com.saveourtool.save.backend.event
+
+import com.saveourtool.save.backend.service.NotificationService
+import com.saveourtool.save.backend.service.UserDetailsService
+import com.saveourtool.save.domain.Role
+import com.saveourtool.save.entities.Notification
+import com.saveourtool.save.entities.cosv.VulnerabilityMetadata
+import com.saveourtool.save.entities.vulnerability.VulnerabilityStatus
+import org.springframework.context.event.EventListener
+import org.springframework.stereotype.Component
+
+/**
+ * A vulnerability metadata listener for sending notifications.
+ */
+@Component
+class VulnerabilityMetadataListener(
+    private val userDetailsService: UserDetailsService,
+    private val notificationService: NotificationService,
+) {
+    /**
+     * @param vulnerability new VulnerabilityMetadata
+     */
+    @EventListener
```

Comment:
So this event is triggered when vulnerabilityMetadata is changed?

Reply:
Only in the place where we subscribed up for it. Right now I have only added the subscription to the create method. Then I want to add to change the vulnerability. But now we have the same status for both new and changed vulnerabilities, so we cannot distinguish between them.

Reply:
Where is the subscription? I do not see it in this PR

Reply:
it: `applicationEventPublisher.publishEvent(newMetadata)`

Reply:
missed that, yes, sorry